pandas/core/indexes/base.py

```diff
@@ -896,18 +896,19 @@ def _formatter_func(self):
         """
         return default_pprint
 
-    def _format_data(self, name=None):
+    def _format_data(self, name=None) -> str_t:
         """
         Return the formatted data as a unicode string.
         """
 
         # do we want to justify (only do so for non-objects)
-        is_justify = not (
-            self.inferred_type in ("string")
-            or (
-                self.inferred_type == "categorical" and is_object_dtype(self.categories)
-            )
-        )
+        is_justify = True
+
+        if self.inferred_type == "string":
+            is_justify = False
+        elif self.inferred_type == "categorical":
+            if is_object_dtype(self.categories):  # type: ignore
```

Comment:
the error here is `pandas\core\indexes\base.py:910: error: "Index" has no attribute "categories"`. should this special case be in CategoricalIndex instead of in the base Index class?